base/src/main/java/butter/droid/base/providers/media/VodoProvider.java

```diff
@@ -77,7 +79,22 @@ protected Call enqueue(Request request, com.squareup.okhttp.Callback requestCall
                         "Mozilla/5.0 (Linux; U; Android %s; %s; %s Build/%s) AppleWebkit/534.30 (KHTML, like Gecko) PT/%s",
                         Build.VERSION.RELEASE, LocaleUtils.getCurrentAsString(), Build.MODEL, Build.DISPLAY,
                         versionName)).build();
-        return super.enqueue(request, requestCallback);
+        Call call = super.enqueue(request, requestCallback);
+
+        synchronized (ongoingCalls) {
```

Comment:
In some scenarios this may produce a bottleneck if we do multiple request for different kind of movies to the same provider instance (as popcorn does). The OkHttp Dispatcher handle the calls internally and all that calls are accesible via `queuedCalls()` or `runningCalls()`, In my opinion the cancel by tag mechanism is better and that logic should be moved to the Media or Base provider and keep the providers as clean as possible.